internal/future/statestore/redis.go

```diff
@@ -94,29 +96,208 @@ func redisCloser(closer func() error) {
 	}
 }
 
-// CreateTicket creates a new Ticket in the state storage. This method fails if the Ticket already exists.
-func (rb *redisBackend) CreateTicket(context.Context, pb.Ticket, int) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+// CreateTicket creates a new Ticket in the state storage. If the id already exists, it will be overwritten.
+func (rb *redisBackend) CreateTicket(ctx context.Context, ticket *pb.Ticket, ttl int) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Error("failed to connect to redis")
+		return err
+	}
+	defer redisConn.Close()
+
+	redisConn.Send("MULTI")
+	key := ticket.Id
+	value, err := proto.Marshal(ticket)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   key,
+			"error": err.Error(),
+		}).Error("failed to marshal the ticket proto")
+		return err
+	}
+
+	err = redisConn.Send("SET", key, value)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   key,
+			"cmd":   "SET",
+			"error": err.Error(),
+		}).Error("failed to set the value for ticket in state storage")
+		return err
+	}
+
+	if ttl > 0 {
```

Comment:
It's not clear why `ttl` is a separate parameter in this method. Also should that just be time.Duration instead of an int? It's not clear what the units are.